demo/imgui/CMakeLists.txt

```diff
@@ -0,0 +1,33 @@
+# Copyright (C) 2018 by Godlike
+# This code is licensed under the MIT license (MIT)
+# (http://opensource.org/licenses/MIT)
+
+cmake_minimum_required (VERSION 3.1)
+
+project(PegasusDemo_imgui)
+
+#GLBINDING
```

Comment:
I would not go as far as putting it and GLFW here since IMGUI is not the sole reason those dependencies exist.

Reply:
Any suggestion how to separate imgui compilation from Pegasus build flags? `g++` was failing to build in `Release` mode because of some unused variable in `third_party/imgui`. The value was used in assertion which is ignored in `Release` build making the variable unused.

The issue here was that `GL/gl3w.h` is including `demo/Gl.hpp` which in turn includes both `glbinding` and `glfw`. In complication to that, `glbinding_api.h` is created in a build directory derived from current project build directory so the subsequent `include` of it failed unless `add_subdirectory` was callled from the bottom level of project hierarchy ().

Since library is built explicitly `STATIC`, it proxies all of its linked libraries to those who are linking it.

Reply:
I could also rename it from `demo/imgui` to `demo/dependencies` to make intent more clear

Reply:
I think the rename will do the job.